lib/src/components/xmark.dart

```diff
@@ -0,0 +1,60 @@
+import 'package:flutter/material.dart';
+
+/// XMark Icon
+class XMark extends StatelessWidget {
+  /// size of the canvas to draw a cross on.
+  final Size size;
+
+  /// color of the lines.
+  final Color color;
+
+  /// width of cross lines.
+  final double lineWidth;
+
+  /// Constructor
+  const XMark({
+    Key? key,
+    required this.size,
+    required this.color,
+    required this.lineWidth,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return CustomPaint(
+      size: size,
+      painter: _CrossPainter(lineWidth: lineWidth, color: color),
+    );
+  }
+}
+
+class _CrossPainter extends CustomPainter {
+  final double lineWidth;
+  final Color color;
+
+  /// constructor
+  _CrossPainter({required this.lineWidth, required this.color});
```

Comment:
Nitpick/personal preference:
```suggestion
  _CrossPainter({required this.lineWidth, required this.color,});
```